framework/Models/PostProcessors/Validations/Probabilistic.py

```diff
@@ -0,0 +1,145 @@
+        specifying input of cls.
+    """
+    specs = super(Probabilistic, cls).getInputSpecification()
+    #specs.addSub(metricInput)
+    return specs
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'POSTPROCESSOR Probabilistic'
+    self.dynamicType = ['static','dynamic'] #  for now only static is available
+    self.acceptableMetrics = ["CDFAreaDifference", "PDFCommonArea"] #  acceptable metrics
+    self.name = 'Probabilistic'
+    # self.pivotParameter = None
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+
+
+
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it loads the
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
+    pivotParameter = self.pivotParameter
+    names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+    if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+      if 'dynamic' not in self.dynamicType: #self.model.dataType:
+        self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+
+    if pivotParameter:
+      if len(inputIn['Data'][0][-1]['time']) != len(list(evaluation.values())[0]):
+        self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+      if pivotParameter not in evaluation:
+        evaluation[pivotParameter] = inputIn['Data'][0][-1]['time']
+    return evaluation
+
+  ### utility functions
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    names = kwargs.get('dataobjectNames')
+    outputDict = {}
+    for feat, targ in zip(self.features, self.targets):
+      featData = self._getDataFromDataDict(datasets, feat, names)
+      targData = self._getDataFromDataDict(datasets, targ, names)
+      for metric in self.metrics:
+        name = "{}_{}_{}".format(feat.split("|")[-1], targ.split("|")[-1], metric.estimator.name)
```

Comment:
Hm, could this feat.split("|")[-1] ever result in overlapping names? (As in if we have bar|baz and bat|baz they will end up overwriting in the dictionary)

Reply:
Tho' this did exist in the previous version, so without renaming the name, it may not be fixable.